tests/incident/test_views.py

```diff
@@ -24,101 +40,318 @@ def tearDown(self):
     def test_validate_event_type_for_incident_acknowledge_raises_validation_error(self):
         incident = StatefulIncidentFactory(source=self.source)
         viewfactory = RequestFactory()
-        request = viewfactory.get(f"/api/v1/incidents/{incident.pk}/events/")
+        request = viewfactory.get(path=f"/api/v1/incidents/{incident.pk}/events/")
         request.versioning_scheme = versioning.NamespaceVersioning()
         request.version = "v1"
         view = EventViewSet()
         view.request = request
         with self.assertRaises(serializers.ValidationError):
-            view.validate_event_type_for_incident(Event.Type.ACKNOWLEDGE, incident)
+            view.validate_event_type_for_incident(event_type=Event.Type.ACKNOWLEDGE, incident=incident)
 
 
 class IncidentViewSetV1TestCase(APITestCase):
-    @classmethod
-    def setUpClass(cls):
+    def setUp(self):
         disconnect_signals()
         source_type = SourceSystemTypeFactory()
-        cls.user = SourceUserFactory()
-        cls.source = SourceSystemFactory(type=source_type, user=cls.user)
-
-    def setUp(self):
+        self.user = SourceUserFactory()
+        self.source = SourceSystemFactory(type=source_type, user=self.user)
+        self.admin = AdminUserFactory()
         self.client.force_authenticate(user=self.user)
 
-    @classmethod
-    def tearDownClass(cls):
+    def teardown(self):
         connect_signals()
 
     def add_incident(self, description="incident"):
-        data = {
-            "start_time": "2022-05-24T13:07:29.254Z",
-            "end_time": "2022-05-24T13:07:29.254Z",
-            "description": description,
-            "level": 1,
-            "tags": [{"tag": "a=b"}],
-        }
-        response = self.client.post("/api/v1/incidents/", data, format="json")
-        self.assertEqual(response.status_code, 201)  # Created
-        return response.data["pk"]
+        incident = StatefulIncidentFactory(source=self.source, description=description)
+        tag = TagFactory(key="a", value="b")
+        IncidentTagRelationFactory(incident=incident, tag=tag)
+        return incident.pk
+
+    def add_event(self, incident_pk, description="event", type=Event.Type.OTHER):
+        return EventFactory(incident_id=incident_pk, description=description, type=type).pk
+
+    def add_acknowledgement(self, incident_pk):
+        event_pk = self.add_event(incident_pk=incident_pk, type=Event.Type.ACKNOWLEDGE)
+        return AcknowledgementFactory(event_id=event_pk).pk
```

Comment:
First change `add_acknowledgement` so it returns the Ack created by the factory directly
```suggestion
    def add_acknowledgement(self):
        return AcknowledgementFactory()
  ```
The Ack factory creates an event with the eventfactory, and the eventcatory creates an incident, so all these things now exist.
Then just use these values directly (see other comment where i suggest change for a specific test

Reply:
Why did I not think of that? Thanks!